### PR TITLE
JBIDE-24963-6 - add new testclasses for CDI 2.0

### PR DESCRIPTION
**Pull request by zcervink**
**JBIDE-24963-6 - Add the following testclasses:**
- BeanDiscoveryInExplicitArchivesTest20.class
 - BeanDiscoveryInImplicitArchivesTest20.class
 - BeanParametersAnnotationTest20.class

Signed-off-by: Zbynek Cervinka <zcervink@redhat.com>

**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/76/
**Jira:** https://issues.jboss.org/browse/JBIDE-24963

please review @jkopriva

### CODE REVIEW

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
testPR

**Comment by zcervink** [comment]
testPR

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInImplicitArchivesTest.java [hunk 1]
```diff
@@ -60,21 +48,6 @@
...
-			new WaitUntil(new EditorHasValidationMarkers(ed));
-		} catch (WaitTimeoutExpiredException ex){
-			fail("There is supposed to be warning present");
-		}
-		List<Marker> markers = ed.getMarkers();
-		assertEquals(1, markers.size());
-		Marker validation = markers.get(0);
-		assertEquals(validation.getLineNumber(), 7);
-		assertTrue(validation.getText().contains("No bean is eligible for injection"));
+		super.validationOfBeanDiscoveryInImplicitArchives();
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInImplicitArchivesTest.java]: Add CDI11 pattern to the class name. Also, In new CDI20 test classes you have specified CDIVersion, add it to the CDI 11 classes as well, of course, value will be for 1.2 or 1.1.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java [hunk 2]
```diff
@@ -25,23 +35,54 @@
...
+		try{
+			new WaitUntil(new EditorHasValidationMarkers(ed));
+		} catch (WaitTimeoutExpiredException ex){
+			fail("There is supposed to be warning present");
+		}
+		List<Marker> markers = ed.getMarkers();
+		assertEquals(1, markers.size());
+		Marker validation = markers.get(0);
+		assertEquals(validation.getLineNumber(), 8);
+		assertTrue(validation.getText().contains("No bean is eligible for injection"));
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java]: We will need to improve the message/problem check. Here was checked only problem's short message, now we have to distinguish between CDI11 and CDI20 - JSR-346 and JSR-365 specs. Some validator would solve the issue well.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java [hunk 3]
```diff
@@ -25,23 +35,54 @@
...
+		try{
+			new WaitUntil(new EditorHasValidationMarkers(ed));
+		} catch (WaitTimeoutExpiredException ex){
+			fail("There is supposed to be warning present");
+		}
+		List<Marker> markers = ed.getMarkers();
+		assertEquals(1, markers.size());
+		Marker validation = markers.get(0);
+		assertEquals(validation.getLineNumber(), 7);
+		assertTrue(validation.getText().contains("No bean is eligible for injection"));
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java]: Same as above.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java [hunk 4]
```diff
@@ -41,6 +44,9 @@
 	AllAssignableDialogTestCDI20.class,
 	AssignableDialogFilterTestCDI20.class,
 	AsYouTypeValidationTestCDI20.class,
+	BeanDiscoveryInExplicitArchivesTest20.class,
```
[the remark this answers is not in the document]
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/CDI20SuiteTest.java]: Fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java (hunk 2 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java]: Fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java (hunk 3 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInArchivesTemplate.java]: Fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInImplicitArchivesTest.java (hunk 1 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInImplicitArchivesTest.java]: CDIVersion specification add to the CDI 11 classes.

Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInImplicitArchivesTest.java]: Is it necessary to rename **super.validationOfBeanDiscoveryInImplicitArchives();** and **super.validationOfBeanDiscoveryInExplicitArchives();** by adding **CDI11** and **CDI20**? Both CDI11 and CDI20 test classes are using the same test methods collected in template class (BeanDiscoveryInArchivesTemplate). The will be duplicity - either there will be 2 same functions **validationOfBeanDiscoveryInExplicitArchives();** and 2 same function for **validationOfBeanDiscoveryInImplicitArchives();** in the template class or there will be duplicity in the classes, that extends the template class.

Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi11/BeanDiscoveryInImplicitArchivesTest.java]: CDI11 pattern added to the class name.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+			fail("There is supposed to be warning present");
+		}
+
+		ProblemsView pv = new ProblemsView();
+		pv.open();
+		List<Problem> problems = pv.getProblems(ProblemType.ALL, new MarkerTypeMatcher("CDI Problem"));
+		assertEquals(1, problems.size());
+		Problem validation = problems.get(0);
+		assertTrue(validation.getLocation().contains("7"));
+		assertTrue(validation.getDescription().contains("No bean is eligible for injection"));
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java]: Here also should be particular JSR number in warning messaged checked.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * 
+ * @author zcervink
+ *
+ */
+public class BeanParametersAnnotationTemplate extends CDITestBase {
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java]: move the class from foo.fooo.foooo.annotation.cdi11 to foo.fooo.foooo.annotation.template package. Same for any other template class that is misplaced in *.cdiXX folder. It should be place in its own template package within * (discovery, annotation) package. Check the other classes ;). This also count for any other PRs.

Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java]: Moved.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java (hunk 5 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTemplate.java]: Fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi20/BeanDiscoveryInExplicitArchivesTestCDI20.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+import org.eclipse.reddeer.eclipse.ui.perspectives.JavaEEPerspective;
+import org.eclipse.reddeer.junit.annotation.RequirementRestriction;
+import org.eclipse.reddeer.junit.requirement.matcher.RequirementMatcher;
+import org.eclipse.reddeer.requirements.jre.JRERequirement.JRE;
+import org.eclipse.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.eclipse.reddeer.requirements.server.ServerRequirementState;
+import org.jboss.ide.eclipse.as.reddeer.server.family.ServerMatcher;
+import org.jboss.ide.eclipse.as.reddeer.server.requirement.ServerRequirement.JBossServer;
+import org.jboss.tools.cdi.bot.test.CDITestBase;
+import org.jboss.tools.cdi.bot.test.beansxml.discovery.cdi11.BeanDiscoveryInArchivesTemplate;
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi20/BeanDiscoveryInExplicitArchivesTestCDI20.java]: move template class into its own package as it is being used by both cdi version tests (1.1 and 2.0).

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTest.java [hunk 8]
```diff
@@ -49,7 +31,7 @@
 @JRE(cleanup=true)
 @OpenPerspective(JavaEEPerspective.class)
 @JBossServer(state=ServerRequirementState.PRESENT, cleanup=false)
-public class BeanParametersAnnotationTest extends CDITestBase {
+public class BeanParametersAnnotationTest extends BeanParametersAnnotationTemplate {
```
Comment (odockal) [inline tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTest.java]: Add suffix CDI11 to this test class.

Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/annotation/cdi11/BeanParametersAnnotationTest.java]: Fixed.

File: tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi20/BeanDiscoveryInExplicitArchivesTestCDI20.java (hunk 7 above)
Comment (zcervink) [reply tests/org.jboss.tools.cdi.bot.test/src/org/jboss/tools/cdi/bot/test/beansxml/discovery/cdi20/BeanDiscoveryInExplicitArchivesTestCDI20.java]: Fixed.

**Comment by zcervink** [comment]
**Jenkins:** https://dev-platform-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/cdi20.itests/121/

**Comment by odockal** [comment]
+1. Good work. @jkopriva Merge please. And consider giving me rights to this repo... :)

**Pull request merged**